### PR TITLE
Add configuration to prevent permanent item theft (Covet / Thief) from NPCs

### PR DESCRIPTION
**Pull request by moostoet**
## Description
In Generation 5 onward, Thief and Covet no longer _permanently_ steal items from trainers. From Bulbapedia:

> Covet no longer steals a Trainer's Pokémon's items permanently; however, a wild Pokémon's item is still permanently stolen by the player. 

> Thief no longer steals a Trainer's Pokémon's items permanently; however, a wild Pokémon's item is still permanently stolen by the player. 

However, as it currently stands, this updated behavior was not implemented. As such, this PR implements the following:

- The addition of a variable `B_RETURN_STOLEN_NPC_ITEMS` in `battle.h` that rudimentarily handles generation-based logic for Thief/Covet against NPCs;
- A refactor of the `TryRestoreHeldItem` function that includes the above variable. Individuals can now choose whether or not trainer NPCs are able to have their items be stolen permanently with Covet/Thief, or not.
- The player still keeps all items obtained through Covet/Thief from Wild Pokemon regardless of variable states.

## **People who collaborated with me in this PR**
- Duke ( @Sneed69 ) for some help with the logic

## **Discord contact info**
'viridian.' (including the dot)

### CODE REVIEW

File: src/battle_util.c [hunk 1]
```diff
@@ -10855,14 +10855,21 @@ void TryRestoreHeldItems(void)
...
             lostItem = gBattleStruct->itemLost[i].originalItem;
-            if (lostItem != ITEM_NONE && ItemId_GetPocket(lostItem) != POCKET_BERRIES)
-                SetMonData(&gPlayerParty[i], MON_DATA_HELD_ITEM, &lostItem);  // Restore stolen non-berry items
+
+            // Check if the lost item should be restored
+            if ((lostItem != ITEM_NONE || returnNPCItems)
+                && ItemId_GetPocket(lostItem) != POCKET_BERRIES)
+            {
+                SetMonData(&gPlayerParty[i], MON_DATA_HELD_ITEM, &lostItem);
+            }
```
Comment (AlexOn1ine) [inline src/battle_util.c:10872]: I think the if statement is small enough for one line and we don't use brackets for a one line unless it is in an if-else-block that has ifs(else) blocks with multiply lines. 

Also the NPC item is still stolen if the user used up a berry in the battle.

**Review by AlexOn1ine — CHANGES_REQUESTED** [review]
Something like this usually goes to upcoming. Master is for bug fixes.

File: src/battle_util.c (hunk 1 above)
Comment (AlexOn1ine) [reply src/battle_util.c:10872]: mon triggers it's own berry during battle, then uses covet to steal opponents held item (a potion in this case). After the battle ends the mon still holds the potion.

**Comment by moostoet** [comment]
Making a separate PR for this implementation as the new PR implements the changes from a forked upcoming branch.

**Pull request closed without merging**